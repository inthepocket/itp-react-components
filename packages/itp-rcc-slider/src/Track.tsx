import * as React from 'react';
import { GetTrackProps } from 'react-compound-slider/Tracks/Tracks';
import { SliderItem } from 'react-compound-slider';

export interface TrackStyles {
  Track?: string;
}

export interface TrackProps {
  key?: string;
  getTrackProps: GetTrackProps;
  prefixCss?: string;
  styles?: TrackStyles;
  source: SliderItem;
  target: SliderItem;
}

const Track = ({
  getTrackProps,
  prefixCss,
  source,
  styles,
  target,
}: TrackProps) => {
  const className = styles ? styles.Track : `${prefixCss}__Track`;

  return (
    <div
      style={{
        left: `${source.percent}%`,
        width: `${target.percent - source.percent}%`
      }}
      className={className}
      {...getTrackProps()}
    />
  );
};

export default Track;
