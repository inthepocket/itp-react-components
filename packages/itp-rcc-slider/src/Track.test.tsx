import * as React from 'react';
import { create } from 'react-test-renderer';
import Track from './Track';

test('Track: default', () => {
  const tree = create((
    <Track
      getTrackProps={() => null}
      prefixCss="Slider"
      source={{ id: 'foo:bar:source', value: 75, percent: 75 }}
      target={{ id: 'foo:bar:target', value: 100, percent: 100 }}
     />
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});

test('Track: css module', () => {
  const tree = create((
    <Track
      getTrackProps={() => null}
      source={{ id: 'foo:bar:source', value: 75, percent: 75 }}
      styles={{
        Track: 'cssModuleTrack',
      }}
      target={{ id: 'foo:bar:target', value: 100, percent: 100 }}
     />
  ))
  .toJSON();

  expect(tree).toMatchSnapshot();
});
