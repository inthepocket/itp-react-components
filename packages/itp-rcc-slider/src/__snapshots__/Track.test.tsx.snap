// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Track: css module 1`] = `
<div
  className="cssModuleTrack"
  style={
    Object {
      "left": "75%",
      "width": "25%",
    }
  }
/>
`;

exports[`Track: default 1`] = `
<div
  className="Slider__Track"
  style={
    Object {
      "left": "75%",
      "width": "25%",
    }
  }
/>
`;
